oauth2client/client.py

```diff
@@ -873,6 +910,44 @@ def _do_revoke(self, http_request, token):
     if self.store:
       self.store.delete()
 
+  def _refresh_scopes(self, http_request):
+    """Refreshes the list of authorized scopes from the OAuth2 provider.
+
+    Args:
+      http_request: callable, a callable that matches the method signature of
+        httplib2.Http.request, used to make the revoke request.
+    """
+    self._do_refresh_scopes(http_request, self.access_token)
+
+  def _do_refresh_scopes(self, http_request, token):
+    """Refreshes the list of authorized scopes from the OAuth2 provider.
+
+    Args:
+      http_request: callable, a callable that matches the method signature of
+        httplib2.Http.request, used to make the refresh request.
+      token: A string used as the token to identify the credentials to the
+        provider.
+
+    Raises:
+      Error: When refresh fails, indicating the the access token is invalid.
+    """
+    logger.info('Refreshing scopes')
+    query_params = {'access_token': token, 'fields': 'scope'}
+    token_info_uri = _update_query_params(self.token_info_uri, query_params)
+    resp, content  = http_request(token_info_uri)
+    if resp.status == 200:
+      d = json.loads(content)
+      self.scopes = util.string_to_scopes(d.get('scope', ''))
+    else:
+      error_msg = 'Invalid response %s.' % resp.status
```

Comment:
Is there a case where ``resp.status`` can be a tuple? This code is identical to the error handling code above in ``_do_revoke``.